### ISSUE TITLE
Get the pattern for the current route

### ISSUE THREAD

**Issue by rightaway** [alexmingoia/koa-router#500]
If I do `ctx.url` I can get the actual URL like `/users/123/456`, but I'd like to suggest a feature where we can get the router pattern that was matched like `/users/:idA/:idB` for the current route. Something like `ctx.router.pattern`.

**Comment by DinaYakovlev** [comment]
you can get the matched route by using: ctx._matchedRoute

**Comment by uniquenode** [comment]
@DY6229 do you know how to get the matched route when using nested routers? In that case the info dissapears

**Comment by DinaYakovlev** [comment]
@ChrisLahaye Yes I encountered the same issue. couldn't find any simple solution to this, except for getting the route by-myself using the `router.match()` function.
for example lets say you have the url: `/pets/{pet_id}/test`
you can see that `ctx._matchedRoute` will look something like this: `/pets/{pet_id}(.*)`
the  `/pets/{pet_id}` is your main route, so you can conclude that `/test` will be the sub route. 
using `router.match('/test', method)` you can get all possible routes and from there to build your own matched route.
You can look at my full implementation here:
 https://github.com/Zooz/prometheus-api-metrics/blob/07c0c2a364ab1978a86198630464f1a58a905d15/src/koa-middleware.js#L33

**Comment by uniquenode** [comment]
@DinaYakovlev thank you for your reply. I just stopped using nested routes and attached multiple routers to my webserver.
